blog post so far (#131)
boa-dev/boa-dev.github.io: blog/2024-02-29-boa-release-18/index.md

## Summary

Boa v0.18 is now available! After 7 months of development we are very happy to present you the latest
release of the Boa JavaScript engine. Boa makes it easy to embed a JS engine in your projects, and
you can even use it from WebAssembly. See the [about](/about) page for more info.

In this release, our conformance has grown from 79.36% to 85.03% in the official ECMAScript Test Suite
(Test262). This means we now pass **3,550** more tests than in the previous version. Moreover, our
amount of ignored tests decreased from **9,496** to **1,391** thanks to all the new builtins we have
implemented for this release.

You can check the full list of changes [here][changelog], and the full information on conformance
[here][conformance].

> This release also marks a major update to the design of our website! We're looking forward to adding some
> additional features to learn more about Boa and writing more blog posts about using Boa and breaking
> down implementation details in the future. Subscrive to our [RSS feed][feed] if you're interested in
> staying up to date!

Reviewer: ```suggestion
You probably noticed that something seems different... This release marks a major update to the design of our website! We'd like to thank [@ZackMitkin](https://github.com/ZackMitkin) for being the one that started the
work on this nifty redesign.
We're planning to add some additional pages to learn more about the APIs that
Boa exposes.
Additionally, expect some more blog posts from us in the future! We would like
to write about how to use certain APIs, design challenges that we encountered
while developing the engine, and internal implementation details.
Subscribe to our [RSS feed][feed] if you're interested in staying up to date!
```

release of the Boa JavaScript engine. Boa makes it easy to embed a JS engine in your projects, and
you can even use it from WebAssembly. See the [about](/about) page for more info.

In this release, our conformance has grown from 79.36% to 85.03% in the official ECMAScript Test Suite
(Test262). This means we now pass **3,550** more tests than in the previous version. Moreover, our
amount of ignored tests decreased from **9,496** to **1,391** thanks to all the new builtins we have
implemented for this release.

You can check the full list of changes [here][changelog], and the full information on conformance
[here][conformance].

You probably noticed that something seems different... This release marks a major update to the design of our website! We'd like to thank [@ZackMitkin](https://github.com/ZackMitkin) for being the one that started the work on this nifty redesign.
We're planning to add some additional pages to learn more about the APIs that
Boa exposes.
Additionally, expect some more blog posts from us in the future! We would like
to write about how to use certain APIs, design challenges that we encountered
while developing the engine, and internal implementation details.

> Subscribe to our [RSS feed][feed] if you're interested in staying up to date!

Reviewer: ```suggestion
We're planning to add some additional pages to learn more about the APIs that Boa exposes.
Additionally, expect some more blog posts from us in the future! We would like
to write about how to use certain APIs, design challenges that we encountered
while developing the engine, and internal implementation details.
Subscribe to our [RSS feed][feed] if you're interested in staying up to date!
```

## Summary

Boa v0.18 is now available! After 7 months of development we are very happy to present you the latest
release of the Boa JavaScript engine. Boa makes it easy to embed a JS engine in your projects, and
you can even use it from WebAssembly. See the [about](/about) page for more info.

In this release, our conformance has grown from 79.36% to 85.03% in the official ECMAScript Test Suite
(Test262). This means we now pass **3,550** more tests than in the previous version. Moreover, our
amount of ignored tests decreased from **9,496** to **1,391** thanks to all the new builtins we have
implemented for this release.

You can check the full list of changes [here][changelog], and the full information on conformance
[here][conformance].

> You probably noticed that something seems different... This release marks a major update to the design of our website! We'd like to thank [@ZackMitkin](https://github.com/ZackMitkin) for being the one that started the
> work on this nifty redesign.

Reviewer: ```suggestion
You probably noticed that something seems different... This release marks a major update to the design of our website! We'd like to thank [@ZackMitkin](https://github.com/ZackMitkin) for being the one that started the work on this nifty redesign.
```

You can check the full list of changes [here][changelog], and the full information on conformance [here][conformance].

Moreover, this big release was partly possible thanks to those who have [supported us](https://opencollective.com/boa). Thanks to

> funds we've recieved we have been able to renew our domain name, renumerate members of the team who have worked on the features released, utilise servers for benchmarking. If you wish to sponsor Boa, you can do so by donating to our [open collective][collective]. You can also check

Reviewer: ```suggestion
funds we've received we have been able to renew our domain name, remunerate members of the team who have worked on the features released, and discuss the possibility of using dedicated servers for benchmarking. If you wish to sponsor Boa, you can do so by donating to our [open collective][collective]. You can also check
```

[conformance]: https://boajs.dev/boa/test262/
[collective]: https://opencollective.com/boa
[easy_issues]: https://github.com/boa-dev/boa/issues?q=is%3Aopen+is%3Aissue+label%3AE-Easy
[first_issues]: https://github.com/boa-dev/boa/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22

## Highlights

### We're on [test262.fyi](https://test262.fyi/)!

Thanks to the amazing work of [CanadaHonk](https://twitter.com/CanadaHonk), Boa is now listed on
[test262.fyi](https://test262.fyi/). https://test262.fyi/ runs a nightly build of Boa, along with other engines, and updates the results every day. This is using the tip of the main branch of Boa alongside the [latest test262](https://github.com/tc39/test262) changes pushed to their main branch.

This is a great achievement for us and we are very proud to be listed alongside other major JavaScript engines. It should be much easier for users to compare Boa's conformance tests with other engines.

Below is a screenshot showing Boa being 83% conformance with the latest tests and to the right statistics on the time taken to performa a full test run in comparison to other engines.

![img](./conformance_dark.webp)

### Temporal

> The majority of our work over the past few months has been on the Temporal API. The Temporal API is a new set of built-in objects and functions that provide a modern, ECMAScript language standard for date and time manipulation. The Temporal API is designed to be a more modern replacement for the `Date` object, and it provides a more feature-rich and flexible API for working with dates and times.

Reviewer: ```suggestion
A lot of work has been put over the past few months on the Temporal API. The Temporal API is a new set of built-in objects and functions that is designed to be a more modern replacement for the `Date` object, providing a more feature-rich and flexible API for working with dates and times.
```

A lot of the team members were focused on different parts of the engine, so I don't think it's descriptive to say that we put the majority of our effort on Temporal.

> It is currently a [stage 3 proposal](https://tc39.es/proposal-temporal/docs/) and we are working alongside the TC39 champions to put together a solid implementation. As Temporal is such a large implementation, we have done most of the work outside of Boa so that it can be used in other projects. This work can be found in the [temporal_rs](https://github.com/boa-dev/temporal/) repository.

Reviewer: ```suggestion
It is currently a [stage 3 proposal](https://tc39.es/proposal-temporal/docs/) and we are working alongside the TC39 champions to put together a solid implementation. Since Temporal is such an extensive specification, we have done most of the work outside of Boa so that it can be used in other projects. This work can be found in the [temporal_rs](https://github.com/boa-dev/temporal/) repository.
```

[easy][easy_issues] or [good first issues][first_issues].

[changelog]: https://github.com/boa-dev/boa/blob/v0.18/CHANGELOG.md
[conformance]: https://boajs.dev/boa/test262/
[collective]: https://opencollective.com/boa
[easy_issues]: https://github.com/boa-dev/boa/issues?q=is%3Aopen+is%3Aissue+label%3AE-Easy
[first_issues]: https://github.com/boa-dev/boa/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22

## Highlights

### We're on [test262.fyi](https://test262.fyi/)!

Thanks to the amazing work of [CanadaHonk](https://twitter.com/CanadaHonk), Boa is now listed on
[test262.fyi](https://test262.fyi/). https://test262.fyi/ runs a nightly build of Boa, along with other engines, and updates the results every day. This is using the tip of the main branch of Boa alongside the [latest test262](https://github.com/tc39/test262) changes pushed to their main branch.

This is a great achievement for us and we are very proud to be listed alongside other major JavaScript engines. It should be much easier for users to compare Boa's conformance tests with other engines.

> Below is a screenshot showing Boa being 83% conformance with the latest tests and to the right statistics on the time taken to performa a full test run in comparison to other engines.

Reviewer: ```suggestion
Below is a screenshot showing Boa at 83% conformance with the latest tests and to the right statistics on the time taken to perform a full test run in comparison to other engines.
```
